POST: 6/11 | "What's new" click-to-parse-post-v3 (#17584)
newrelic/docs-website: src/content/whats-new/2024/06/whats-new-06-11-click-to-parse.md

---
title: 'Streamline log parsing'
summary: 'Use our new options to generate ingest parsing or query time parsing rules'
releaseDate: '2024-06-11'
learnMoreLink: 'https://newrelic.com/blog/how-to-relic/click-to-parse'
getStartedLink: 'https://youtu.be/tvK6MlkvD6Y?si=5klygPBzQyY5qsI8'
---

While the majority of valuable parsing information is automatically gathered during the ingestion of logs into the New Relic platform, building additional queries can be labor-intensive. Logs parsing just became easier for both types of parsing at New Relic: 

> * Ingest parsing: These are rules that create permanent changes in your logs and are based on Grok rules. In the past, you had to create your own Grok rules, but now we can create these complex rules for you with our NRAI assistant.  

Reviewer: ```suggestion
* Ingest parsing: These are rules that create permanent changes in your logs and are based on New Relic AI  rules. In the past, you had to create your own New Relic AI  rules, but now we can create these complex rules for you with our New Relic AI assistant.  
```

Reviewer: I don't believe we use NRAI as a shorthand, but I could be wrong. I would check out our usage dictionary to be sure. If we do allow NRAI, I would:

1. Use New Relic AI (NRAI) on the first instance
2. Update all subsequent references to New Relic AI/Grok to say NRAI

Reviewer 2: This is actually meant to be Grok!

> * Query time parsing: We generate temporary parsing rules written in NRQL as you make log selections in our UI.

Reviewer: maybe bold the text before the `:`

Reviewer 2: Done

> ## Ingest parsing with AI assistance

Reviewer: ```suggestion
## Ingest parsing with AI assistance [#ingest-parsing-AI]
```

> We are expanding our public preview for our AI-powered ingest to allow you to parse your logs using automatically generated Grok rules. New Relic AI can help you create and troubleshoot parsing rules by:

Reviewer: ```suggestion
We are expanding our public preview for our AI-powered ingest to allow you to parse your logs using automatically generated New Relic AI rules. New Relic AI can help you create and troubleshoot parsing rules by:
```

Reviewer 2: Keeping Grok!

* **Suggesting parsing rules**: When you create a parsing rule in the UI, ask NRAI to create the parsing rule for you. 

> * **Explaining parsing rules**: If you don't understand the parsing rule our AI suggests, you can ask AI to explain the rule and suggest an even more efficient one if available. 

Reviewer: New Relic AI vs just referring to 'AI'

![A screenshot showing NRAI enabled.](./images/pic1.png "A screenshot showing NRAI enabled.")

> <figcaption>Ask AI to help you generate Grok patterns for ingest parsing.</figcaption>

Reviewer: ```suggestion
<figcaption>Ask New Relic AI to help you generate New Relic AI patterns for ingest parsing.</figcaption>
```

Reviewer: a little repetitive/redundant but def want to remove mentions of grok

Reviewer 2: This is actually Grok! It tripped me up too. It's a logs thing.

To try the new AI-powered log parsing capability, join our public preview:

> 1. Go to the user menu at the bottom left of the New Relic UI.

Reviewer: recommend using this pattern for orderred lists:

```
1.
1.
1.
1.
1.
```

2. Click the **Features** tab.

> 3. Scroll down to the **New Relic** AI section and click **Enable**.

Reviewer: ```suggestion
3. Scroll down to the **New Relic AI** section and click **Enable**.
```

Reviewer 2: Done

4. **Give Us Feedback** by following the option under the Help pulldown.

> ## Query time parsing rules

Reviewer: anchor tag

Reviewer 2: thanks!

> New Relic's query time parsing feature allows you to select any string from your log file and then parse out repeatable values, reducing the time it takes to create queries. We do all the work for you by creating the NRQL query as you make selections from your logs in the UI.

Reviewer: per style guide, we don't use New Relic in the possessive. Recommend changing sentence to something like:

> Our query time parsing feature...

> To get started:

Reviewer: i would reframe this section as an additional content, rather than a "get started", which implies procedures to actually start using a feature/functionality

Reviewer: also convert to an unordered list, as i don't think anyone needs to read this content in this specific order
